core/read.c

```diff
@@ -893,6 +893,35 @@ static int isom_read_stbl( lsmash_file_t *file, isom_box_t *box, isom_box_t *par
     return isom_read_children( file, box, stbl, level );
 }
 
+static int isom_read_SA3D( lsmash_file_t *file, isom_box_t *box, isom_box_t *parent, int level )
+{
+    ADD_BOX( SA3D, isom_audio_entry_t );
+    lsmash_bs_t *bs = file->bs;
+
+    SA3D->version = lsmash_bs_get_byte( bs );
+    uint8_t temp8 = lsmash_bs_get_byte( bs );
+    SA3D->head_locked_stereo = (temp8 >> 7) & 1;
+    SA3D->ambisonic_type = temp8 & 0x7F;
+    SA3D->ambisonic_order = lsmash_bs_get_be32( bs );
+    SA3D->ambisonic_channel_ordering = lsmash_bs_get_byte( bs );
+    SA3D->ambisonic_normalization = lsmash_bs_get_byte( bs );
+    SA3D->num_channels = lsmash_bs_get_be32( bs );
+    for( uint64_t pos = lsmash_bs_count( bs ); pos < box->size && SA3D->channel_map.entry_count < SA3D->num_channels; pos = lsmash_bs_count( bs ) )
```

Comment:
`while` might be more readable? Though, if this sort of `for` is use elsewhere in the codebase, I suppose this would be preferred.